stakingModel/test/GoodMarketMaker.test.js

```diff
@@ -60,16 +51,109 @@ contract(
       expect(onecDAIReturn.toNumber() / 100).to.be.equal(10000); //0.0001 cdai is 1 gd, so for 1eth you get 10000 gd (divide by 100 to account for 2 decimals precision)
     });
 
+    it("should update reserve ratio by yearly rate", async () => {
+      const expansion = await marketMaker.reserveRatioDailyExpansion();
+      expect(expansion.toString()).to.be.equal("999388834642296000000000000");
+      await marketMaker.expandReserveRatio(cDAI.address);
+      const daytwoRR = await marketMaker.reserveTokens(cDAI.address);
+      expect(daytwoRR["1"].toString()).to.be.equal("999388");
+      await marketMaker.expandReserveRatio(cDAI.address);
+      const daythreeRR = await marketMaker.reserveTokens(cDAI.address);
+      expect(daythreeRR["1"].toString()).to.be.equal("998777");
+    });
+
     it("should calculate mint UBI correctly for 8 decimals precision", async () => {
       const gdPrice = await marketMaker.currentPrice(cDAI.address);
-      const toMint = await marketMaker.shouldMint(cDAI.address, "100000000");
+      const toMint = await marketMaker.calculateMintInterest(cDAI.address, "100000000");
       const expectedTotalMinted = 10 ** 8 / gdPrice.toNumber(); //1cdai divided by gd price;
       expect(expectedTotalMinted).to.be.equal(10000); //1k GD since price is 0.0001 cdai for 1 gd
       expect(toMint.toString()).to.be.equal(
         (expectedTotalMinted * 100).toString()
       ); //add 2 decimals precision
     });
 
+    it("should be able to calculate and update bonding curve gd balance based on oncoming cDAI and the price stays the same", async () => {
+      const priceBefore = await marketMaker.currentPrice(cDAI.address);
+      await marketMaker.mintInterest(
+        cDAI.address,
+        web3.utils.numberToHex(1e8)
+      );
+      let reserveToken = await marketMaker.reserveTokens(cDAI.address);
+      let gdSupply = reserveToken.gdSupply;
+      let reserveSupply = reserveToken.reserveSupply;
+      let reserveRatio = reserveToken.reserveRatio;
+      await marketMaker.initializeToken(
```

Comment:
need to call mintInterest and not initialize

Reply:
actually probably need to block initialiezToken for a token that is already initialized